### ISSUE TITLE
Cop idea: minimize number of negations in a condition

### ISSUE THREAD

**Issue by vlad-pisanov** [rubocop/rubocop#11399]
Currently, `Style/NegatedIf` (and `Style/NegatedUnless`) recommend `if x` over `unless !x`, which is good because negations are less natural for the reader.

What might be useful is a cop that goes a step further and recommends `if x || y` over `unless !x && !y`. That is, a cop that reduces the number of negations in a boolean expression as much as possible by leveraging DeMorgan's Rule.

In its basic form, it could look for conjunctions (or disjunctions) where _all_ terms are negated:

```ruby
# bad (all terms negated)
unless !x && !y && !z

# good (all terms positive)
if x || y || z
```

A more complex version could look for conjunctions (or disjunctions) where there are _more negations_ than positives:
```ruby
# bad (3 negations, 1 positive)
unless !x || !y || !z || w

# good (3 positives, 1 negation)
if x && y && z && !w
```

Does this sound reasonable? 🤔

**Comment by fatkodima** [comment]
I had a similar suggestion a while ago - https://github.com/rubocop/rubocop/issues/8879. Just need to implement 😄

**Comment by fatkodima** [comment]
Valid.
